ndk-glue/src/lib.rs

```diff
@@ -279,10 +279,10 @@ unsafe extern "C" fn on_input_queue_destroyed(
     activity: *mut ANativeActivity,
     queue: *mut AInputQueue,
 ) {
+    wake(activity, Event::InputQueueDestroyed);
     let input_queue = InputQueue::from_ptr(NonNull::new(queue).unwrap());
     input_queue.detach_looper();
     *INPUT_QUEUE.write().unwrap() = None;
```

Comment:
For safety I wonder if we should assert here that `queue == INPUT_QUEUE.ptr()`. That also means the lock should be moved up so that the entire comparison + `detach_looper` + `= None` assignment happens while it is held:

```suggestion
    wake(activity, Event::InputQueueDestroyed);
    let mut write_guard = INPUT_QUEUE.write().unwrap();
    let input_queue = InputQueue::from_ptr(NonNull::new(queue).unwrap());
    assert_eq!(queue, write_guard.as_ref().unwrap().ptr().as_ptr());
    input_queue.detach_looper();
    *write_guard = None;
```